oml/metrics/embeddings.py

```diff
@@ -148,114 +115,79 @@ def __init__(
         self.metrics_to_exclude_from_visualization = ["fnmr@fmr", "pcf", *metrics_to_exclude_from_visualization]
         self.verbose = verbose
 
-        keys_to_accumulate = [self.embeddings_key, self.is_query_key, self.is_gallery_key, self.labels_key]
-        keys_to_accumulate += [INDEX_KEY]  # todo 522: remove it after we make "indices" not optional in .update_data()
-        if self.categories_key:
-            keys_to_accumulate.append(self.categories_key)
-        if self.sequence_key:
-            keys_to_accumulate.append(self.sequence_key)
-        if self.extra_keys:
-            keys_to_accumulate.extend(list(extra_keys))
-
-        self.keys_to_accumulate = tuple(set(keys_to_accumulate))
-        self.acc = Accumulator(keys_to_accumulate=self.keys_to_accumulate)
-
-    def setup(self, num_samples: int) -> None:  # type: ignore
-        self.distance_matrix = None
-        self.mask_gt = None
+        self._acc_embeddings_key = "__embeddings"
+        self.acc = Accumulator(keys_to_accumulate=(self._acc_embeddings_key,))
+
+    def setup(self, num_samples: Optional[int] = None) -> None:  # type: ignore
+        self.retrieval_results = None
         self.metrics = None
+        self.metrics_unreduced = None
 
-        self.acc.refresh(num_samples=num_samples)
+        num_samples = num_samples if num_samples is not None else len(self.dataset)
```

Comment:
Difficult logic 🤔 

On the one hand, metric gets the dataset in constructor, so it always has to be `len(self.dataset)`. On the other hand, such logic is invalid for DDP setup...

Reply:
I have a mixed feeling... If it's optional, it looks redundant to call this function without args in the code. With optional it's more natural to call it implicitly during the first call of `update` or explicitly in the DDP code.

Reply:
not sure how I can implement this...